config/config.go

```diff
@@ -50,7 +50,7 @@ func loadFiles(config interface{}, fnames ...string) error {
 			return err
 		}
 
-		if err := yaml.Unmarshal(data, config); err != nil {
+		if err := yaml.UnmarshalStrict(data, config); err != nil {
```

Comment:
Can we make this configurable with a default of on? For better or worse we depend on this in a few places already (e.g. collector).